build.gradle

```diff
@@ -0,0 +1,331 @@
+	testImplementation files("$jarDir/apache_commons/commons-lang-2.4.jar")
+	testImplementation files("$jarDir/apache_commons/velocity-1.7.jar")
+	testImplementation files("$jarDir/arangodb/arangodb-java-driver-6.7.2.jar")
+	testImplementation files("$jarDir/arangodb/velocypack-2.4.0.jar")
+	testImplementation files("$jarDir/bytebuddy/byte-buddy-1.9.10.jar")
+	testImplementation files("$jarDir/bytebuddy/byte-buddy-agent-1.9.10.jar")
+	testImplementation files("$jarDir/equalsverifier/equalsverifier-3.1.10.jar")
+	testImplementation files("$jarDir/jsemver/java-semver-0.9.0.jar")
+	testImplementation files("$jarDir/kbase/auth2/kbase-auth2test-0.2.4.jar")
+	testImplementation files("$jarDir/mockito/mockito-core-3.0.0.jar")
+	testImplementation files("$jarDir/objenesis/objenesis-2.6.jar")
+}
+
+// download the KBase jars if they aren't already present
+task downloadJars {
+	doLast {
+		def dir = file(downloadDir)
+		if (!dir.exists()) {
+			dir.mkdirs()
+		}
+		def url = "https://github.com/kbase/jars/raw/master/lib/jars/kbase"
+		def jars = [
+			"auth/kbase-auth-0.4.4.jar",
+			"auth2/kbase-auth2test-0.2.4.jar",
+			"common/$kbaseCommonJar",
+			"handle/AbstractHandleClient-1.0.0.jar",
+			"kidl/kbase-kidl-parser-1409261812-7863aef.jar",
+			"sample/SampleServiceClient-0.1.1.jar",
+			"shock/shock-client-0.1.0.jar"
+		]
+		jars.each { jar ->
+			def jarName = jar.split("/")[-1]
+			def file = file("$downloadDir/$jarName")
+			if (!file.exists()) {
+				ant.get(src: "$url/$jarName", dest: "$downloadDir/$jarName")
+			}
+		}
+	}
+}
+
+// Custom java project layout
+sourceSets {
+	main {
+		java {
+			srcDir src
+			exclude "**/test/**"
+		}
+	}
+	test {
+		java {
+			srcDir src
+		}
+		resources {
+			srcDir src
+			include "**/*.properties"
+			include "**/*.cfg"
+			include "**/*.spec"
+			include "**/*.instance.*"
+			include "**/instance.*"
+			include "**/*.instance"
+			include "**/*.html"
+			include "**/*.css"
+			include "**/*.gif"
+			include "**/*.js"
+			include "**/*.png"
+			include "**/*.txt"
+			include "**/*.weirdsuffix"
+		}
+	}
+}
+
+tasks.withType(JavaCompile) {
+	classpath = configurations.allDeps
+	options.compilerArgs = ["-source", "11", "-target", "11", "-g"]
+}
+
+task clientJar(type: Jar, dependsOn: compileJava) {
+	from sourceSets.main.output
+	include "us/kbase/workspace/*.class"
+	exclude "us/kbase/workspace/WorkspaceServer.class"
+	include "us/kbase/common/service/*.class"
+	exclude "us/kbase/common/service/JsonServer*"
+	archiveFileName = clientJarFile
+	destinationDirectory = file("$dist/client")
+}
+
+task serviceJar(type: Jar, dependsOn: compileJava) {
+	from sourceSets.main.output
+	archiveFileName = serviceJarFile
+	destinationDirectory = file("$dist")
+}
+
+task buildJars {
+	dependsOn serviceJar, clientJar
+}
+
+task unzipJar(type: Copy) {
+	from zipTree("$jarDir/kbase/common/$kbaseCommonJar")
+	into "$tempunpack"
+}
+
+// by default, the files go into build/docs/javadoc
+tasks.javadoc {
```

Comment:
## Build the Javadocs

This task is run by the Gradle Java plugin. It compiles the `main` source set, adds classes from the `kbase-common` jar, and then generates javadocs for a select set of classes. The resulting documentation can be found in the directory `build/docs/javadoc`.

### to test:
run `ant clean`
run `ant javadoc`, which will generate documentation in the directory `docs/javadoc`
run `gradle clean javadoc`
run `diff -r docs build/docs`
Note any differences